app/controllers/patients_controller.rb

```diff
@@ -26,8 +31,14 @@ def edit
     end
 
     def create
+        doctor = Doctor.find_by(name: params[:patient]["doctors_name"])
+        if(doctor.nil?)
+            doctor = Doctor.new
```

Comment:
Do not create a row in doctor's table independently. ActiveRecord allows you to associate the doctor with patient during creation. See an example of this in the PhonesController#create method here - https://github.com/virmani/patient-tracker/blob/master/app/controllers/phones_controller.rb#L7